core/filterconstraint.cpp

```diff
@@ -0,0 +1,939 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "filterconstraint.h"
+#include "dive.h"
+#include "divesite.h"
+#include "errorhelper.h"
+#include "gettextfromc.h"
+#include "qthelper.h"
+#include "tag.h"
+#include "trip.h"
+#include "subsurface-time.h"
+#include <QDateTime>
+
+// We use the units enum only internally.
+// Therefore define it here, not in the header file.
+enum filter_constraint_units {
+	FILTER_CONSTRAINT_NO_UNIT = 0,
+	FILTER_CONSTRAINT_LENGTH_UNIT,
+	FILTER_CONSTRAINT_TEMPERATURE_UNIT,
+	FILTER_CONSTRAINT_WEIGHT_UNIT,
+	FILTER_CONSTRAINT_VOLUMETRIC_FLOW_UNIT
+};
+
+static struct type_description {
+	filter_constraint_type type;
+	const char *s;			// untranslated string that, which will be written to the log
+	const char *translated;
+	bool has_string_mode;
+	bool has_range_mode;
+	bool is_star_widget;
```

Comment:
that one is obvious, I think I can guess about the range mode, but what's the string mode (sure, I could read the source, but I always love nice explanations in the struct declaration...)

Reply:
I added more comments, even though they are somewhat redundant.